### PR TITLE
CMake: Migrate macOS and Windows to the OpenSSL formula

### PR DESCRIPTION
**Pull request by alessandrogario**
~~The formula should work, but I still have to add a script to install Perl on the Windows CI machine~~

Ready for review!

### CODE REVIEW

File: libraries/cmake/formula/openssl/CMakeLists.txt [hunk 1]
```diff
@@ -38,29 +28,133 @@ function(opensslMain)
...
+        zlib-dynamic
+        enable-cms
+        "--with-zlib-include=${thirdparty_zlib_INCLUDE_DIRS}"
+        "--with-zlib-lib=${thirdparty_zlib_LIBRARIES}"
+        darwin64-x86_64-cc
+        enable-ec_nistp_64_gcc_128
+        -fPIC
+    )
+
+    set(build_command
```
Comment (theopolis) [inline libraries/cmake/formula/openssl/CMakeLists.txt]: Can we add a note referencing OpenSSL's documentation about this. Without context this looks weird. :D

Comment (theopolis) [reply libraries/cmake/formula/openssl/CMakeLists.txt]: Maybe we should also add a check in CMake for this path? For folks building on different versions of macOS or if this path / Xcode changes we might run into difficult-to-debug situations.

File: libraries/cmake/formula/openssl/CMakeLists.txt [hunk 2]
```diff
@@ -38,29 +28,133 @@ function(opensslMain)
...
 
-    INSTALL_COMMAND
+    set(optional_install_command
       make install
+    )
+
+  elseif("${CMAKE_SYSTEM_NAME}" STREQUAL "Windows")
+    unset(perl_executable_path CACHE)
+    find_program(perl_executable_path "perl.exe"
+      PATHS "C:\\Strawberry\\perl\\bin"
```
Comment (theopolis) [inline libraries/cmake/formula/openssl/CMakeLists.txt]: 🍓

**Comment by theopolis** [comment]
We need to update the build documentation to mention the new expectations for Windows: install Perl and Nasm. We might want to give a heads up to the Windows compiler team, whom regularly builds osquery.

**Comment by alessandrogario** [comment]
> We need to update the build documentation to mention the new expectations for Windows: install Perl and Nasm. We might want to give a heads up to the Windows compiler team, whom regularly builds osquery.

I'm updating the documentation next. I also want to restore the Azure config changes you made in the other PR to show the formula log because it was really useful!

**Comment by theopolis** [comment]
Looks good, @alessandrogario, I will let you decide if you want to squash or land the series of individual commits.

**Pull request merged**